components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/ProxyFacetSupport.java

```diff
@@ -53,44 +56,68 @@
     extends FacetSupport
     implements ProxyFacet
 {
-  public static final String CONFIG_KEY = "proxy";
-
-  private URI remoteUrl;
+  @VisibleForTesting
+  static final String CONFIG_KEY = "proxy";
+
+  @VisibleForTesting
+  static class Config
+  {
+    // TODO: Normalize URI ends with "/"?
+    @NotEmpty
```

Comment:
I won't forget :-P  Just trying to decide if this is should be done after configuration is loaded, or with @JsonProperty and/or if the "remoteUrl" here should be a String and converted to URI with normalization.